cartography/intel/okta/users.py

```diff
@@ -153,19 +155,23 @@ def _load_okta_users(neo4j_session, okta_org_id, user_list, okta_update_tag):
     )
 
 
-def sync_okta_users(neo4j_session, okta_org_id, okta_update_tag, okta_api_key):
+def sync_okta_users(neo4j_session, okta_org_id, okta_update_tag, okta_api_key, sync_state):
     """
     Sync okta users
     :param neo4j_session: Session with Neo4j server
     :param okta_org_id: Okta organization id to sync
     :param okta_update_tag: The timestamp value to set our new Neo4j resources with
     :param okta_api_key: Okta API key
+    :param sync_state: Okta sync state
     :return: Nothing
     """
 
     logger.debug("Syncing Okta users")
     user_client = _create_user_client(okta_org_id, okta_api_key)
     data = _get_okta_users(user_client)
-    users = transform_okta_user_list(data)
+    users_data, user_ids = transform_okta_user_list(data)
```

Comment:
As above, I might have missed something but the `user_ids` array might not be needed because objects in `users_data` already have IDs in them.

Reply:
see comment above. big waste of memory to keep all that data around. We keep it short to id which is all we need,

Reply:
I won't block on this, (and correct me if I'm wrong or missing something big) but I think it's actually more memory to keep track of an additional array.

`users_data` is a list of dicts which each contain an `id` field.  Lists and dict parameters are passed in Python as object reference, so there are no additional copies of the ID being made - we just need to access the reference to the ID, which can save memory instead of creating an additional `user_ids` list.

If perf ends up being a problem in this module then we will revisit and optimize this call in the future.